app/src/main/java/com/hover/stax/channels/ChannelListViewModel.java

```diff
@@ -42,9 +42,10 @@ public ChannelListViewModel(Application application) {
 		super(application);
 		repo = new DatabaseRepo(application);
 		loadChannels();
+		initSelectedChannels();
 		loadSims();
-		simHniList = Transformations.map(sims, this::getHnis);
-		simCountryList = Transformations.map(sims, this::getSimCountries);
+		simHniList = Transformations.map(sims, this::getHnisAndSubscribeToEachOnFirebase);
```

Comment:
Nice. I do tend to go for shorter names over fully descriptive names cause I am lazy, but this is a good example of why that is bad and this is much better.